aiounifi/models/device.py

```diff
@@ -834,6 +849,14 @@ def port_table(self) -> list[TypedDevicePortTable]:
         """List of ports and data."""
         return self.raw.get("port_table", [])
 
+    @property
+    def speedtest_status(self) -> TypedDeviceSpeedtestStatus | None:
+        """Speedtest status."""
+        value = self.raw.get("speedtest-status")
+        if value is None:
+            return None
+        return cast(TypedDeviceSpeedtestStatus, value)
```

Comment:
This is the bit I'm not sure about. If I leave it as just `return self.raw.get("speedtest-status")`, it complains about the return type not matching. 🤷🏻

Reply:
That is because the typeddict can't handle the '-' and instead its added as a '_'.

Check system_stats in the device which has the same issue

Reply:
Is it ok as-is then? 🙂

Reply:
Convert to a tuple instead of a typed dict? 🤔 Happy to make that change.

```suggestion
        data = self.raw.get("speedtest-status")
        if data is None:
            return None
        return (
            data.get("latency", 0),
            data.get("rundate", 0),
            data.get("runtime", 0),
            data.get("status_download", 0),
            data.get("status_ping", 0),
            data.get("status_summary", 0),
            data.get("status_upload", 0),
            data.get("xput_download", 0),
            data.get("xput_upload", 0),
        )
```

Or do you mean removing the `None` check? 🤔 `speedtest-status` won't be included for most devices, so that might not be possible.

Reply:
```suggestion
        if (value := self.raw.get("speedtest-status")) is None:
            return None
        return cast(TypedDeviceSpeedtestStatus, value)
```
Sorry, that was a bad example as it didn't do what I thought, small suggestion on improving your original PR

Reply:
Ah, didn't know you could do that. Looks much cleaner!

Reply:
There's a lot of nice additions to the python language over the last few releases.